packages/kotti-ui/source/kotti-toaster/create-toaster.test.ts

```diff
@@ -0,0 +1,460 @@
+import { describe, expect, it, vitest } from 'vitest'
+
+import { createToaster } from './create-toaster'
+
+const DEBUG: boolean = false
+
+const createAnimationFrameMock = () => {
+	let interval: Timer | null = null
+	return {
+		getIsRunning: () => interval !== null,
+		start: (update: () => void) => {
+			if (interval)
+				throw new Error(
+					'Could not start animation frame, already running. This is likely a bug.',
+				)
+
+			if (DEBUG) console.log('animation-frame-mock: start')
+			interval = globalThis.setInterval(() => {
+				if (DEBUG) console.log('animation-frame-mock: update')
+				update()
+			}, 5)
+		},
+		stop: () => {
+			if (DEBUG) console.log('animation-frame-mock: stop')
+			if (interval) {
+				globalThis.clearInterval(interval)
+				interval = null
+			}
+		},
+	}
+}
+
+describe('createToaster', () => {
+	it('returns things', () => {
+		const toaster = createToaster({
+			animationFrame: createAnimationFrameMock(),
+		})
+
+		expect(toaster).toEqual({
+			_internal_pls_dont_touch: expect.anything(),
+			abort: expect.anything(),
+			show: expect.anything(),
+			withOptions: expect.anything(),
+		})
+	})
+
+	describe('.abort()', () => {
+		it('can abort a toast', async () => {
+			const toaster = createToaster()
+
+			const toast = toaster.show({ duration: 1, text: 'test' })
```

Comment:
set duration to 1000 or null or something, too short duration may provoke odd failure where the toast gets deleted before it can be aborted.